client/main.lua

```diff
@@ -1,268 +1,227 @@
-local VehicleSpawn5 = vector3(1247.0718994141, -344.65634155273, 69.08)
-local DriverWep = "WEAPON_MICROSMG" 											-- << the weapon the driver is to be equipped with
-local NavWep = "WEAPON_MICROSMG" 												-- << the weapon the guard should be equipped with
-local TimeToBlow = 30 * 1000 													-- << bomb detonation time after planting, default 20 seconds
+
 local PickupMoney = 0
 local BlowBackdoor = 0
-local SilenceAlarm = 0
 local PoliceAlert = 0
-local PoliceBlip = 0
 local LootTime = 1
 local GuardsDead = 0
-local prop
 local lootable = 0
 local BlownUp = 0
-local TruckBlip
-local transport
 local MissionStart = 0
 local warning = 0
-local VehicleCoords = nil
-local dealer
 local PlayerJob = {}
+local dealerZone = nil
+local TruckBlip = nil
+local transport = nil
+local dealer = nil
+local VehicleCoords = nil
+local bag = nil
+local prop = nil
 local pilot = nil
 local navigator = nil
 
-RegisterNetEvent('QBCore:Client:OnPlayerLoaded', function()
-    QBCore.Functions.GetPlayerData(function(PlayerData)
-        PlayerJob = PlayerData.job
+-- Functions
+
+local function createKeyListener(key, eventType, event)
+    local listening = true
+    CreateThread(function()
+        while listening do
+            if IsControlJustPressed(0, key) then
+                if eventType == 'client' then
+                    TriggerEvent(event)
+                elseif eventType == 'server' then
+                    TriggerServerEvent(event)
+                end
+                exports['qb-core']:KeyPressed()
+                listening = false
+            end
+            Wait()
+        end
     end)
-end)
-
-RegisterNetEvent('QBCore:Client:OnJobUpdate', function(JobInfo)
-    PlayerJob = JobInfo
-end)
-
-function hintToDisplay(text)
-    SetTextComponentFormat("STRING")
-    AddTextComponentString(text)
-    DisplayHelpTextFromStringLabel(0, 0, 1, 20000)
 end
 
--- Ped spawn and mission accept
-Citizen.CreateThread(function()
-    while true do
-        Citizen.Wait(0)
-        local plyCoords = GetEntityCoords(PlayerPedId(), false)
-        local dist = #(plyCoords - vector3(MissionMarker.x, MissionMarker.y, MissionMarker.z))
-
-        if dist <= 25.0 then
-            if not DoesEntityExist(dealer) then
-                RequestModel("s_m_y_dealer_01")
-                while not HasModelLoaded("s_m_y_dealer_01") do
-                    Wait(10)
+function CheckVehicleInformation()
+    if IsVehicleStopped(transport) then
+        if GuardsDead == 1 then
+            if not IsEntityInWater(PlayerPedId()) then
+                RequestAnimDict('anim@heists@ornate_bank@thermal_charge_heels')
+                while not HasAnimDictLoaded('anim@heists@ornate_bank@thermal_charge_heels') do
+                    Wait(50)
+                end
+                local x,y,z = table.unpack(GetEntityCoords(PlayerPedId()))
+                prop = CreateObject(GetHashKey('prop_c4_final_green'), x, y, z+0.2,  true,  true, true)
```

Comment:
```lua
joaat('prop_c4_final_green')
```

Reply:
- Correct the ped task after accepting mission as ped continually walks
- Rear door distance check for planting bomb without target
- Rear door target rather than the vehicle for use with target